### PR TITLE
Initial release with get_url functionality

### PR DESCRIPTION
**Pull request by kowalewskijan**
This is an internal PR to review the initial release before we create a PR to mainline.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+def read(fname):
+    return open(os.path.join(os.path.dirname(__file__), fname)).read()
+
+setup(name='stpm',
+      version='0.1',
+      description='SymbiFlow Tools Package Manager',
+      url='http://github.com/antmicro/symbiflow-tools-package-manager',
+      author='Antmicro',
+      author_email='jkowalewski@antmicro.com',
```
Comment (acomodi) [inline setup.py]: I would change this to symbiflow authors probably

```suggestion
      author='SymbiFlow Authors',
      author_email='symbiflow@lists.librecores.org',
```

File: stpm/__init__.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2020  The SymbiFlow Authors.
+#
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier:	ISC
+
+import json, requests
```
Comment (acomodi) [inline stpm/__init__.py]: IMO it is better to separate each import in its own line

File: stpm/__init__.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2020  The SymbiFlow Authors.
```
Comment (acomodi) [inline stpm/__init__.py:4]: Python scripts should have the utf-encoding and python shebang as well. (I think this is not getting tested by the CI scripts?).

e.g. :
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
#
# Copyright ...
```

File: stpm/__init__.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier:	ISC
+
+import json, requests
+
+
+# For now we use only `symbiflow-arch-defs` and download tarballs
+def get_url(project='symbiflow-arch-defs', build_name='install'):
```
Comment (acomodi) [inline stpm/__init__.py]: Probably we need to rename this to something like `get_latest_artifact_url`.

**Review by acomodi — CHANGES_REQUESTED** [review]
Some comments in.

In general looks good, I think there are some things missing that could be worth adding:

- Python auto-formatter (to test in CI)
- Addition of a console entry point in the setup, such that we can call this functionality from command line.

**Comment by acomodi** [comment]
I think that ones the comments are fixed we can then proceed by closing this PR and opening one directly upstream.

**Comment by kowalewskijan** [comment]
I resolved the comments. Closing here, I'll open the PR in mainline repository.

**Pull request closed without merging**